pkg/connector/custom_role.go

```diff
@@ -0,0 +1,328 @@
+	groups, resp, err := client.Group.ListGroups(ctx, qp)
+	if err != nil {
+		return nil, nil, fmt.Errorf("okta-connectorv2: failed to fetch groups from okta: %w", err)
+	}
+
+	reqCtx, err := responseToContext(token, resp)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return groups, reqCtx, nil
+}
+
+func (o *customRoleResourceType) Grants(
+	ctx context.Context,
+	resource *v2.Resource,
+	token *pagination.Token,
+) ([]*v2.Grant, string, annotations.Annotations, error) {
+	var rv []*v2.Grant
+	_, page, err := parsePageToken(token.Token, resource.Id)
+	if err != nil {
+		return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to parse page token: %w", err)
+	}
+
+	_, bag, err := unmarshalSkipToken(token)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	if bag.Current() == nil {
+		// Push onto stack in reverse
+		bag.Push(pagination.PageState{
+			ResourceTypeID: resourceTypeGroup.Id,
+		})
+		bag.Push(pagination.PageState{
+			ResourceTypeID: resourceTypeUser.Id,
+		})
+	}
+
+	adminFlags, respCtx, err := listAdministratorRoleFlags(ctx, o.client, token, page)
+	if err != nil {
+		// We don't have permissions to fetch role assignments, so return an empty list
+		if errors.Is(err, errMissingRolePermissions) {
+			return nil, "", nil, nil
+		}
+
+		return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to list users: %w", err)
+	}
+
+	switch bag.ResourceTypeID() {
+	case resourceTypeGroup.Id:
+		pageGroupToken := "{}"
+		for pageGroupToken != "" {
+			groupToken := &pagination.Token{
+				Token: pageGroupToken,
+			}
+			bagGroups, pageGroups, err := parsePageToken(groupToken.Token, resource.Id)
+			if err != nil {
+				return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to parse page token: %w", err)
+			}
+
+			qp := queryParams(groupToken.Size, pageGroups)
+			groups, respGroupCtx, err := listGroups(ctx, o.client, groupToken, qp)
+			if err != nil {
+				return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to list groups: %w", err)
+			}
+
+			nextGroupPage, _, err := parseResp(respGroupCtx.OktaResponse)
+			if err != nil {
+				return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to parse response: %w", err)
+			}
+
+			err = bagGroups.Next(nextGroupPage)
+			if err != nil {
+				return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to fetch bag.Next: %w", err)
+			}
+
+			for _, group := range groups {
+				groupId := group.Id
+				roles, _, err := listGroupAssignedRoles(ctx, o.client, groupId, nil)
+				if err != nil {
+					return nil, "", nil, err
+				}
+
+				for _, role := range roles {
+					if role.Status == roleStatusInactive || role.AssignmentType != "GROUP" || role.Type != roleTypeCustom {
+						continue
+					}
+
+					// It's a custom role. We need to match the label to the display name
+					if role.Label == resource.GetDisplayName() {
+						rv = append(rv, roleGroupGrant(groupId, resource))
```

Comment:
I think these also should do a `GrantExpandable` 

```
btipling@C1s-MacBook-Pro baton-okta % baton grants --resource-type custom-role
ID                                                                   | Resource Type | Resource    | Entitlement             | Principal
custom-role:cr0kym5plv1BEzuk25d7:assigned:user:00u49lomo2cv0c7us5d7  | Custom Role   | test-custom | test-custom Role Member | Joe Palmer
custom-role:cr0kym5plv1BEzuk25d7:assigned:group:00g40ajcvnqLyW47T5d7 | Custom Role   | test-custom | test-custom Role Member | admin | super 2
``` 

I would have expected all the users in the group to be listed there and I think our customers expect that too when they run access reviews

Reply:
Not quite sure what do you mean by GrantExpandable.. May be you could give me more context about it.